src/System.CommandLine/Binding/ModelBinder.cs

```diff
@@ -37,208 +26,210 @@ internal ModelBinder(IValueDescriptor valueDescriptor)
             }
-
-            return values.SingleOrDefault()?.Value;
+            var constructorAndArgs = GetBestConstructorAndArgs(bindingContext);
+            var constructor = constructorAndArgs.Constructor;
+            var boundValues = constructorAndArgs.BoundValues;
+            bool nonDefaultsUsed = constructorAndArgs.NonDefaultsUsed;
+            return constructor is null
+                ? GetSimpleModelValue(ConstructorArgumentBindingSources, bindingContext)
+                : InstanceFromSpecificConstructor(bindingContext, constructor, boundValues, ref nonDefaultsUsed);
         }
 
-        private bool TryDefaultConstructorAndPropertiesStrategy(
-            BindingContext context,
-            [NotNullWhen(true)] out object? instance)
+        private bool DisallowedBindingType()
         {
-            var constructorDescriptors =
-                ModelDescriptor
-                    .ConstructorDescriptors
-                    .OrderByDescending(d => d.ParameterDescriptors.Count);
-
-            foreach (var constructor in constructorDescriptors)
+            var disallowedTypes = new List<Type>
             {
-                var boundConstructorArguments = GetValues(
-                    ConstructorArgumentBindingSources,
-                    context, 
-                    constructor.ParameterDescriptors,
-                    true);
-
-                if (boundConstructorArguments.Count != constructor.ParameterDescriptors.Count)
-                {
-                    continue;
-                }
-
-                // Found invokable constructor, invoke and return
-                var values = boundConstructorArguments.Select(v => v.Value).ToArray();
-
-                try
-                {
-                    var fromModelBinder = constructor.Invoke(values);
+                typeof(Span<>),
+                typeof(ReadOnlySpan<>)
+            };
+            var type = ModelDescriptor.ModelType;
+            return disallowedTypes
+                    .Any(x => type.IsGenericType && (type.GetGenericTypeDefinition() == x));
+        }
 
-                    UpdateInstance(fromModelBinder, context);
+        private bool ShortCutTheBinding(BindingContext bindingContext)
+        {
+            var explicitTypesToShortcut = new List<Type>
+            {
+                typeof(string)
+            };
+            Type modelType = ModelDescriptor.ModelType;
+            return modelType.IsPrimitive ||
+                   IsNullable(modelType) ||
+                   explicitTypesToShortcut.Contains(modelType);
+
+            static bool IsNullable(Type type)
+            {
+                return type.IsGenericType &&
+                       type.GetGenericTypeDefinition() == typeof(Nullable<>);
+            }
+        }
 
-                    instance = fromModelBinder;
+        private (bool success, object? newInstance, bool anyNonDefaults) GetSimpleModelValue(
```

Comment:
It appears that the `success` and `anyNonDefaults` always have the same value. Would it be cleaner to have this be a bool returning method with an out parameter for the `newInstance`?